cpp/src/parquet/arrow/writer.cc

```diff
@@ -546,6 +554,10 @@ Status GetSchemaMetadata(const ::arrow::Schema& schema, ::arrow::MemoryPool* poo
   std::string schema_as_string = serialized->ToString();
   std::string schema_base64 = ::arrow::util::base64_encode(schema_as_string);
   result->Append(kArrowSchemaKey, std::move(schema_base64));
+
+  // Add any required fields collected by the GeoCrsContext
+  properties.geo_crs_context()->AddProjjsonCrsFieldsToFileMetadata(result.get());
```

Comment:
It is weird to mutate the member variable of a const reference. We can either add GeoCrsContext as a separate parameter or change ArrowWriterProperties to be non-const.